hub/core/version_control/test_version_control.py

```diff
@@ -340,25 +341,371 @@ def test_auto_checkout(local_ds):
 
 def test_auto_commit(local_ds):
     initial_commit_id = local_ds.commit_id
+    # auto commit as head of main branch
     local_ds.checkout("pqr", create=True)
     local_ds.checkout("main")
-    # no auto commit as there was no data
-    assert local_ds.commit_id == initial_commit_id
+    second_commit_id = local_ds.commit_id
+    assert second_commit_id != initial_commit_id
     local_ds.create_tensor("abc")
     local_ds.abc.append(1)
-    # auto commit as there was data
+    # auto commit as head of main again
     local_ds.checkout("xyz", create=True)
     local_ds.checkout("main")
 
-    assert local_ds.commit_id != initial_commit_id
+    assert local_ds.commit_id != second_commit_id
 
     with local_ds:
         local_ds.abc.append(1)
 
-    current_commit_id = local_ds.commit_id
+    third_commit_id = local_ds.commit_id
 
-    # auto commit as there was data
+    # auto commit as head of main again
     local_ds.checkout("tuv", create=True)
     local_ds.checkout("main")
 
-    assert local_ds.commit_id != current_commit_id
+    assert local_ds.commit_id != third_commit_id
+
+
+def test_diff_linear(local_ds, capsys):
```

Comment:
Going forward like to see a more robust approach to test parameters. We can discuss what package in python world exist that allow for parametric tests that obviate the need for hardcoded values.

Reply:
we do have parametrization with different kinds of datasets with our fixtures. This test doesn't need to use s3/gcs datasets though, so just using "local_ds" fixture works

Reply:
I was actually referring to lines 371->, so the test body itself and not the source of the data. Things like `local_ds.xyz.extend([1, 2, 3])`. How about something like:

`def test_diff_linear(test_spec, local_ds, capsys)` where `test_spec` is a dictionary of vars and funcs that generate values for the test body (which should be ideally entirely free of any hardcoded input). For now, you can init the `test_spec` with values but at a later date we can fuzz the code with randomly generated `test_spec`.